connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -460,14 +460,20 @@ private class KeyValueGroupedDatasetImpl[K, V, IK, IV](
   }
 
   override protected def aggUntyped(columns: TypedColumn[_, _]*): Dataset[_] = {
-    // TODO(SPARK-43415): For each column, apply the valueMap func first...
+    val aggCols: Seq[Column] = valueMapFunc match {
```

Comment:
@xupefei why do we need custom handling on the server side? I dislike that because it is something very scala specific that is now part of the public API. Yes we have accidentally done that in the past, but that does not make it a good idea :)...

Can we instead create a projection of the form `select(keys..., fn(col(*)).as("value)"))` and rewrite the agg exprs to use the value column as its input in the `ColumnNodeToProtoConverter` (`col(*)` would have to be rewritten to `col (value)`)?

Reply:
I tried the projection idea solely on the client side, and it seems non trivial.
I'll see if bringing `AppendColumns` to protobuf can help.

Reply:
Thanks Herman!
I tried select/project but it doesn't work due to `*` is not transformed correctly.
But I managed to make this work by using `df.mapPartitions(valueMapFunc)`. Still halfway done but will continue.